core/domain/event_services.py

```diff
@@ -93,6 +94,54 @@ def _handle_event(
                 rule_spec_index, classification_categorization, params,
                 session_id, time_spent_in_secs))
 
+        if feconf.ENABLE_NEW_STATS_FRAMEWORK:
+            feedback_is_useful = (
+                classification_categorization != (
+                    exp_domain.DEFAULT_OUTCOME_CLASSIFICATION))
+
+            stats_models.AnswerSubmittedEventLogEntryModel.create(
+                exploration_id, exploration_version, state_name, session_id,
+                time_spent_in_secs, feedback_is_useful)
+
+            update_params = {
+                'feedback_is_useful': feedback_is_useful
+            }
+            stats_services.update_stats(
+                exploration_id, exploration_version, state_name, cls.EVENT_TYPE,
+                update_params)
+
+
+class ExplorationActualStartEventHandler(BaseEventHandler):
+    """Event handler for recording exploration actual start events."""
+
+    EVENT_TYPE = feconf.EVENT_TYPE_ACTUAL_START_EXPLORATION
+
+    @classmethod
+    def _handle_event(
+            cls, exp_id, exp_version, state_name, session_id):
+        if feconf.ENABLE_NEW_STATS_FRAMEWORK:
```

Comment:
This is too far in. No point even flooding the event queue with events if nothing's going to be done with them. I recommend putting the "if" guard in the controller layer or even in the frontend (move feconf.ENABLE_NEW_STATS_FRAMEWORK to assets/constants.js in that case and prevent the $http call to the backend altogether).

Reply:
Ah, wait -- I guess that impacts tests. I guess it's OK to have the HTTP traffic, maybe just pull it to the backend controller layer for now.

Reply:
@seanlip Well, that still impacts a lot of the backend tests from stats_servicess (not that we can't change it), but does one extra function call make a lot of difference? This is anyway temporary right?

Reply:
I'm quite hesitant nowadays to assume things are temporary. We've had "temporary" things come in in the past that ended up stalling for longer than expected, so let's keep the develop branch as robust as possible and always submit PRs with the assumption that there is no guarantee of future further changes. In this case I think it does make a difference because this "function call" is actually handled by a deferred queue with limited throughput and events passed through it will take up room that other events could use. So I think we should change it.

The reason I retracted the constants.js thing was because you'd still need feconf.ENABLE_NEW_STATS_FRAMEWORK and a guard in the backend anyway, so let's punt that down the road till Milestone 5. (I was actually mistaken in my original comment about tests, sorry -- I had thought that pulling the constant out into constants.js would be pulling it out of the backend altogether, so we could no longer control its value in tests. But this is incorrect since in the backend we would still be reading it from the json file, and we could also add a stub of some kind.)

Reply:
@seanlip That makes sense.

Another reason I kept it in event_services during development was because of:
There are a few events that are newly introduced and a few existing events. For the existing events, we have no choice but to keep this check in the event_services module. Only for the new modules can we move this check to the controller layer. I felt that keeping it all at the same place would achieve uniformity.

Reply:
Noted -- I thought of that too. I think what we should do is to keep things in the event_services module only if necessary, but I think that uniformity here comes at the expense of scarce resource usage, so it's reasonable to not have uniformity in this case.

Reply:
@seanlip Sounds good! I'll move the check to the controller layer for the newly introduced models.

Reply:
Done.